tamr_unify_client/models/dataset/resource.py

```diff
@@ -55,6 +55,25 @@ def attributes(self):
         resource_json = self.client.get(alias).successful().json()
         return AttributeCollection.from_json(self.client, resource_json, alias)
 
+    def create_attribute(self, attribute_info):
+        """Create an Attribute in Unify
+
+        :param dataset: the Unify Dataset to which to add the attribute
+        :type dataset: :class:`tamr_unify_client.models.dataset.resource.Dataset`
+        :param attribute_info: the name and type (and optional description) of the attribute to create, formatted as described in the `Public Docs for Adding an Attribute <https://docs.tamr.com/reference#add-attributes>`_.
```

Comment:
We highlighted the collision of the `config` term during standup today. I think `info` might be too general. Since this would end up being a specification for how to create the attribute, I propose we call it `attribute_spec`.

Whatever naming scheme we agree on, it would be good to fix the [other usages of `config` in the codebase](https://github.com/Datatamer/unify-client-python/search?l=Python&q=config&type=), namely:

- `project_config` -> `project_spec`
- `dataset_config` -> dataset_spec`

Reply:
@mollysacks could you include separate commits for doing the renaming for `project_config` -> `project_spec` and `dataset_config` -> `dataset_spec`? (make sure to change the names in the corresponding tests too!)